### PR TITLE
Make '(initramfs)' log the only indication of the initramfs status

### PR DESCRIPTION
**Pull request by lubaihua33**
There might be "blk_update_request: I/O error." or ".*exception.*" lines in the log for some images, but they might not cause the system to enter into initramfs. Only there is "(initramfs)" line, the system will enter into initramfs. 

We separate initramfs_logs into two variables, filesystem_exception_logs collects the exception logs to show them in the result's 'Message', initramfs_logs is the indication of initramfs.

### CODE REVIEW

File: lisa/features/serial_console.py [hunk 1]
```diff
@@ -156,6 +159,15 @@ def check_initramfs(
         self._node.log.debug("checking initramfs in serial log...")
         content: str = self.get_console_log(saved_path=saved_path, force_run=force_run)
 
+        filesystem_exception_logs = [
```
Comment (dsrivastavv) [inline lisa/features/serial_console.py:163]: [nit] maybe split this into multiple statements

File: lisa/features/serial_console.py [hunk 2]
```diff
@@ -35,15 +35,18 @@ class SerialConsole(Feature):
...
         re.compile(r"^(.*blk_update_request: I/O error.*)$", re.MULTILINE),
         re.compile(r"^(.*exception.*)$", re.MULTILINE),
         re.compile(
             r"^(.*Failure: File system check of the root filesystem failed.*)$",
             re.MULTILINE,
         ),
-        re.compile(r"^(.*\(initramfs\).*)$", re.MULTILINE),
+    ]
+
+    initramfs_patterns: List[Pattern[str]] = [
```
Comment (dsrivastavv) [inline lisa/features/serial_console.py:49]: Can you please add an example pattern matching this regex?

**Review by squirrelsc — DISMISSED** [review]

File: lisa/features/serial_console.py (hunk 1 above)
Comment (lubaihua33) [reply lisa/features/serial_console.py:163]: I think we don't need to split it into multiple statements.

File: lisa/features/serial_console.py (hunk 2 above)
Comment (lubaihua33) [reply lisa/features/serial_console.py:49]: Have added. Thank you.

**Review by LiliDeng — APPROVED** [review]

File: lisa/features/serial_console.py [hunk 3]
```diff
@@ -35,15 +35,19 @@ class SerialConsole(Feature):
...
         re.compile(r"^(.*blk_update_request: I/O error.*)$", re.MULTILINE),
         re.compile(r"^(.*exception.*)$", re.MULTILINE),
         re.compile(
             r"^(.*Failure: File system check of the root filesystem failed.*)$",
             re.MULTILINE,
         ),
-        re.compile(r"^(.*\(initramfs\).*)$", re.MULTILINE),
+    ]
+
+    # (initramfs)
```
Comment (LiliDeng) [inline lisa/features/serial_console.py:48]: if more lines here, that would be good.

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**